llvm/lib/Target/LoongArch/LoongArchISelLowering.h

```diff
@@ -44,6 +44,8 @@ enum NodeType : unsigned {
   ROTR_W,
 
   // unsigned 32-bit integer division
+  DIV_W,
+  MOD_W,
```

Comment:
Do you really need MOD_W? RISC-V doesn't have REMW. We use default promotion for signed remainder and use this isel pattern.

```
def : Pat<(srem (sexti32 (i64 GPR:$rs1)), (sexti32 (i64 GPR:$rs2))),             
          (REMW GPR:$rs1, GPR:$rs2)>;
```

Reply:
This is a split from PR #116764. [^1] It aims to control the extension mode of the input operands for `div.w`/`mod.w` based on the `div32` target feature. [^2] (On LoongArch, if the `div32` target feature is not enabled, the results of `div.w`/`mod.w` operations are unpredictable if the input operands are not sign-extended. This behavior may differ from RISC-V.

[^1]: https://github.com/llvm/llvm-project/pull/116764/files#r1849838395
[^2]: https://github.com/llvm/llvm-project/pull/116764/files#diff-47299c232403370e6e02355713cf717324051be6fbaa48257087ebdf0d46ae37R2841-R2844